internal/controllers/liveness/namespace.go

```diff
@@ -59,21 +70,27 @@ func (c *namespaceController) Reconcile(ctx context.Context, req ctrl.Request) (
 		logger.V(0).Info("deleting recreated namespace")
 		return ctrl.Result{}, nil
 	}
-	if err == nil { // important that this is the GET error
-		return ctrl.Result{}, nil // namespace exists, nothing to do
+	if err == nil {
```

Comment:
What if this error is non-nil but not not-found?

Reply:
That condition is handled right after the get call